src/go/rpk/pkg/cli/cmd/topic/consume_test.go

```diff
@@ -256,13 +287,32 @@ func TestParseConsumeTimestamp(t *testing.T) {
 
 		// For timestamps, we use a 1min bounds because we could have
 		// relative durations from now.
-		if at.Before(test.at) && test.at.Sub(at) > time.Minute ||
-			test.at.Before(at) && at.Sub(test.at) > time.Minute {
-			t.Errorf("#%d: got at %v not within 1min of %v", i, at, test.at)
+		if startAt.Before(test.startAt) && test.startAt.Sub(startAt) > time.Minute ||
+			test.startAt.Before(startAt) && startAt.Sub(test.startAt) > time.Minute {
+			t.Errorf("#%d: got at %v not within 1min of %v", i, startAt, test.startAt)
 		}
 
 		if end != test.end {
 			t.Errorf("#%d: got end %v != exp end %v", i, end, test.end)
 		}
+
+		offset = test.in[l:]
+		if offset == "" || offset == ":" { // requesting from a time onward; e.g., @1m:
+			continue
+		}
+
+		if fromT0 {
+			_, endAt, _, _, _ = parseConsumeTimestamp(offset[1:], startAt)
+		} else {
+			_, endAt, _, _, _ = parseConsumeTimestamp(offset[1:])
+		}
+		if i == 9 { // in case duration:duration
```

Comment:
this is very fragile, I think we want another struct field in the test type, `bothDuration`

Also the case here is fragile -- 24h0m0 prefix, I think we can do some better thing in the test struct/fields itself?

Reply:
> I think we want another struct field in the test type, bothDuration

How does another struct field help for the test? It's actually fragile, but `test.startAt` and `startAt` using `time.Now()` can't be the same time since the last few digits in milliseconds get different.

Reply:
Another struct field, `bothDuration`, could be true only for this case -- and then rather than `i == 9` it could be `if test.bothDuration`

w.r.t. startAt -- we can check if the times are within, say, 10s, then the time range is close enough-ish. That's what I do above on previous-line-259, now-line-290. Not sure if that makes sense?

Reply:
Ah, really appreciate on elaborating with examples. That totally makes sense.

Reply:
fixed in bf66adf

Reply:
I'd assume it's close enough-ish if the subtract duration is within 1 second.

Reply:
Hopefully! 1s can be flaky in slow CI environments, so we may end up bumping this relatively quickly if it ever flakes.